feat: add user status and location tracking with notifications

gor3a opened the pull request:
## Description
This PR implements user status and location tracking functionality for space members, along with Google Chat notifications for status updates.

### Changes
- Added new fields to `space_members` table:
  - `status`: online/offline status
  - `location`: office/remote location
  - `last_status_update_at`: timestamp of last status update
- Implemented status update UI in the space dashboard
- Added Google Chat notifications for status changes
- Added environment variable `VITE_GOOGLE_CHAT_WEBHOOK_URL` for webhook configuration

### Features
- Users can now set their status (online/offline) and location (office/remote)
- Status updates are broadcasted to Google Chat
- Status UI shows current status and location
- Location field is disabled when status is offline

### Technical Details
- Added new database migration for status fields
- Implemented status update mutation in Supabase
- Added notification utilities for Google Chat integration
- Updated types and schemas to support new fields

### Testing
Please test the following:
1. Setting status to online/offline
2. Setting location when online
3. Verifying Google Chat notifications
4. Checking status persistence after page reload


### Screenshots
![image](https://github.com/user-attachments/assets/a5131d46-f013-49b0-8e92-ab01116de34d)
![image](https://github.com/user-attachments/assets/1cc4753f-88dc-4105-b64d-98d647e3e471)


### Configuration
Add the following to your `.env` file:
`VITE_GOOGLE_CHAT_WEBHOOK_URL`

atitoa93 commented on src/lib/notifications/webhook.ts:
This action should be triggered via a Supabase Edge Function to securely handle and protect the webhook secret.

gor3a replied:
@atitoa93 Agree! I added a new commit, Please review it

gor3a committed:
refactor: move Google Chat webhook to Supabase Edge Function

- Move webhook logic from client to Supabase Edge Function
- Add Edge Function configuration and implementation
- Remove client-side webhook URL environment variable
- Update Supabase client version to 2.24.3
- Add proper error handling and CORS support

gor3a committed:
chore: update .gitignore and add Google Chat webhook environment example

- Add .env file for Supabase functions to .gitignore
- Create .env.example for Google Chat webhook configuration

gor3a committed:
chore: update .gitignore to exclude all .env files and remove Google Chat webhook environment file

- Modify .gitignore to ignore all .env files in Supabase functions
- Delete the specific .env file for Google Chat webhook configuration

gor3a committed:
rebase: new changes of vite

atitoa93 commented on src/env.d.ts:
I believe we don't need this file.

atitoa93 commented on supabase/functions/google-chat-webhook/.env.example:
The secret should be removed from here.

atitoa93 commented on supabase/functions/google-chat-webhook/index.ts:
```diff
@@ -0,0 +1,79 @@
+      }
+    )
+  } catch (error) {
+    return new Response(
+      JSON.stringify({ error: error.message }),
```
I double this 👍

atitoa93 commented on .gitignore:
```diff
@@ -26,6 +26,7 @@ dist-ssr
 # Supabase
 .branches
 .temp
+supabase/functions/**/.env
```
To improve maintainability and streamline deployments, I recommend managing all environment variables directly within the supabase folder. This helps prevent scattering variables across the project, making it easier to aggregate them during deployment or CI/CD processes.

gor3a committed:
refactor: improve error handling and env file management

- Add centralized error handling utility for Supabase functions
- Update .gitignore to better handle Supabase env files
- Remove hardcoded webhook URL from example files
- Clean up unused env type definitions
- Add new .env.example for Supabase configuration

atitoa93 reviewed (approved):
👍
